src/main/java/com/vinyl/service/Validator.java

```diff
@@ -0,0 +1,8 @@
+package com.vinyl.service;
+
+import com.vinyl.model.User;
+
+public interface Validator {
```

Comment:
use generics, the validator interface should be able to validate any object, only the specific implementattions should know about User

Reply:
fixed